bindings/node/lib/stateMachine.js

```diff
@@ -305,6 +307,28 @@ module.exports = function(modules) {
       });
     }
 
+    /**
+     * @ignore
+     * Sets the Node TLS options if provided for the specific KMS provider.
+     */
+    async setTlsOptions(kmsProvider, options) {
+      const tlsOptions = this.options.tlsOptions;
+      if (tlsOptions) {
+        const providerTlsOptions = tlsOptions[kmsProvider];
+        if (providerTlsOptions) {
+          if (providerTlsOptions.tlsCertificateKeyFile) {
+            options.cert = await readFile(providerTlsOptions.tlsCertificateKeyFile, { encoding: 'ascii' });
+          }
+          if (providerTlsOptions.tlsCAFile) {
+            options.ca = await readFile(providerTlsOptions.tlsCAFile, { encoding: 'ascii' });
+          }
```

Comment:
totally optional, but since these are fully independent:

```suggestion
          await Promise.all([
            providerTlsOptions.tlsCertificateKeyFile && (async() => {
             options.cert = await readFile(providerTlsOptions.tlsCertificateKeyFile, { encoding: 'ascii' });
            })(),
            providerTlsOptions.tlsCAFile && (async() => {
              options.ca = await readFile(providerTlsOptions.tlsCAFile, { encoding: 'ascii' });
            })()
          ]);
```

also, quoting the Node.js Buffer docs on the `ascii` encoding:

> 'ascii': For 7-bit ASCII data only. When encoding a string into a Buffer, this is equivalent to using 'latin1'. When decoding a Buffer into a string, using this encoding will additionally unset the highest bit of each byte before decoding as 'latin1'. Generally, there should be no reason to use this encoding, as 'utf8' (or, if the data is known to always be ASCII-only, 'latin1') will be a better choice when encoding or decoding ASCII-only text. It is only provided for legacy compatibility.

I'd remove the encoding option entirely and just read it as a Buffer if that's possible, the TLS API accepts it that way anyway and it saves a string decoding/encoding roundtrip.

(i.e. the suggestion above would become:


```suggestion
          await Promise.all([
            providerTlsOptions.tlsCertificateKeyFile && (async() => {
             options.cert = await readFile(providerTlsOptions.tlsCertificateKeyFile);
            })(),
            providerTlsOptions.tlsCAFile && (async() => {
              options.ca = await readFile(providerTlsOptions.tlsCAFile);
            })()
          ]);
```

)

Reply:
I actually changed since our conversation to use `readFileSync` for the file loading but have also subsequently removed the ascii encoding and returned buffers. Do you still want the file reads async?

Reply:
> Do you still want the file reads async?

I mean, unless there is a reason for them *not* to be asynchronous? Synchronous I/O is definitely an antipattern in production code